docs/adr/isomorphism-support.md

```diff
@@ -0,0 +1,77 @@
+# ADR: CommonJS only output
+
+## Status
+
+Proposed
+
+## Submitters
+
+- @kjugi
+- @wesleytodd
+- @ctcpip
+
+## Decision Owners
+
+- @expressjs/express-tc
+
+## Context
+
+The document's objective is to gather all notable comments and thoughts in one place and track potential changes in this topic. We have noticed that it is repeated frequently in many issues from the community and we need to take action.
+
+Based on the comments in the issues we know that some of you might not feel happy about it. We have acknowledged the need and discussion around it touched on multiple scenarios. Including:
+- rethinking the process and exposing both paths for all libraries
+- expose both paths for selected libraries
+- keeping default settings as the main target is on the server
+
+**Why do we need this decision?**
+- To reduce noise and maintain the ultimate answer for all of the issues.
+
+**What problem does it solve or avoid?**
+- General response to community request.
+
+**Are there any existing issues/discussions/pull requests related to this?**
+- https://github.com/pillarjs/router/issues/128
+- https://github.com/expressjs/discussions/issues/297
+- https://github.com/expressjs/express/discussions/6051
+- https://github.com/jshttp/cookie/issues/211
+
+## Decision
+
+During [working session](https://github.com/expressjs/discussions/issues/320) we have decided to not revisit, investigate or discuss this topic further. That means ESM exports won't be available for expressjs as well for pillarjs and jshttp packages.
+
+**What will be done?**
+
+Future issues can be closed with a link to this document.
+
+## Rationale
+
+CommonJS is the default node.js syntax. The JS world moved in the ESM direction as browsers consumed it well, and bundlers could make a tree-shake feature and dynamic imports. There is still a lot of baggage within the ESM itself and the way how we can use it. 
```

Comment:
```suggestion
CommonJS is the default syntax in Node.js. While the JavaScript ecosystem has increasingly moved toward ESM due to its compatibility with browsers, enhanced tree-shaking capabilities, and support for dynamic imports, there are still complexities and challenges associated with ESM. 
```

Reply:
Not sure if I get that part:
> fwiw treeshaking CJS is identical to ESM in capability

based [on this](https://webpack.js.org/guides/tree-shaking/#conclusion) and [this](https://web.dev/articles/commonjs-larger-bundles)

> in order to take advantage of tree shaking, you must...
> - Use ES2015 module syntax (i.e. import and export).

Meanwhile I will accept this suggestion and we can update it again later if needed.